src/pexprs-madlibs.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var common = require('./common');
+var pexprs = require('./pexprs');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+pexprs.PExpr.prototype.generateExample = common.abstract;
+
+pexprs.any.generateExample = function(grammar, examples, inSyntacticContext, actuals) {
+  return {example: String.fromCharCode(Math.floor(Math.random() * 255))};
+};
+
+function categorizeExamples(examples) {
+  var examplesNeeded = examples.filter(function(example) {
+                                  return example.hasOwnProperty('examplesNeeded');
+                                })
+                               .map(function(example) { return example.examplesNeeded; });
+  examplesNeeded = [].concat.apply([], examplesNeeded)
+                            .filter(function(value, index, arr) {
+                              return arr.indexOf(value) === index;
```

Comment:
I also don't get this...is it supposed to be removing duplicates?

Assuming I understand this code correctly, I think it might be clearer to rewrite it all the map/filter stuff as a single `for` loop. If you need to remove duplicates, you can use an object for that (e.g., `var uniqueExamples = {};` -- then later use `Object.keys` to get an array of the de-duplicated examples).